src/main/java/org/broadinstitute/consent/http/service/UserService.java

```diff
@@ -467,32 +469,36 @@ public List<User> findUsersInJsonArray(String json, String arrayKey) {
     return jsonElementList.stream().distinct().map(e -> findUserById(e.getAsInt())).toList();
   }
 
-  public void hasValidActiveERACredentials(Integer userId) {
-    List<LibraryCard> cards = libraryCardDAO.findLibraryCardsByUserId(userId);
-    List<UserProperty> userProperties = findAllUserProperties(userId);
-    boolean hasEraCommonsId = cards.stream().anyMatch(c -> c.getEraCommonsId() != null);
-    if (!hasEraCommonsId) {
-      throw new BadRequestException("User does not have an Era Commons ID");
-    }
-    List<UserProperty> eraStatusProps = userProperties.stream().filter(
-            userProperty -> userProperty.getPropertyKey().equalsIgnoreCase(ERA_STATUS.getValue()))
-        .toList();
-    List<UserProperty> eraExpirationProps = userProperties.stream().filter(
-            userProperty -> userProperty.getPropertyKey()
-                .equalsIgnoreCase(ERA_EXPIRATION_DATE.getValue()))
-        .toList();
-    if (eraStatusProps.size() == 1 && eraExpirationProps.size() == 1) {
-      if (!eraStatusProps.get(0).getPropertyValue().equalsIgnoreCase("true")) {
-        throw new BadRequestException("User does not have an Era Commons ID that is authorized.");
+    public void hasValidActiveERACredentials(User user) {
+      List<LibraryCard> cards = user.getLibraryCards();
+      List<UserProperty> userProperties = findAllUserProperties(user.getUserId());
```

Comment:
The more I look at this, the more I think I would move `properties` out of `User` rather than making it populated in more cases. Unless there are upcoming use cases where `properties` is needed outside of its use in `DarCollection.createUser`.

Reply:
@rushtong , @pshapiro4broad  - so I think we're in agreement.  I'll leave the PR as-is and make a ticket for us to consider something different later.
@pshapiro4broad - the one solid use case where I see the List used is when we build the object to serialize back to the client in findUserWithPropertiesByIdAsJsonObject.  Ironically, we never seem to populate the list in the user object.  The code is necessary in that method because it allows the DAR form in the UI to become aware that the user has a valid token.

Reply:
Ticket: [https://broadworkbench.atlassian.net/browse/DT-1571](https://broadworkbench.atlassian.net/browse/DT-1571)

Reply:
Yes, in agreement on enrichment.

Secondly, yes, the `findUserWithPropertiesByIdAsJsonObject` is the primary method to populate a user which is used quite a bit in the UI. In theory, that method could be refactored to make use of the enriched user. I think it might be odd to try and restrict user properties to the DarCollection - we need user info for admins, DAC members, SOs, etc., who would generally never create a DAR, but could be eRA Commons/RAS users.